### ISSUE TITLE
Location Cannot Be Deleted

### ISSUE THREAD

**Issue by leobg** [snipe/snipe-it#1153]
I am trying to delete a location I no longer use, but snipe-it refuses to do so as it claims there is at least one asset that is associated with this location. I have not found way to run asset report by location (except by showing all assets and filter by location name, which says I have no assets in the questionable location). However if I go to Admin/Locations - the assets column says '1' for that location I am trying to delete. I have gone through every single asset on in the system, none is associated to this location. I have several assets not associated to any location, not sure if that triggers the flag for the location I am trying to delete. I tried querying the database (table assets, column 'rtd_location_id' but this also tells me I have no assets associated to that location). I'm wondering if this is a bug of some sort.

**Comment by snipe** [comment]
@leobg Do you have any users associated with that location? Also check deleted assets to see if they might be associated with the location.

**Comment by leobg** [comment]
No - not a single user is associated with this location. I have few users who have no location associated thought (if that changes anything). In deleted assets - same thing - no deleted asset is associated with this location. No deleted users either. Just double checked all these again.

**Comment by snipe** [comment]
@leobg What fields were you checking? There are a few fields that could be involved

- `rtd_location_id` in `assets`
- `location_id` in `users`
- `location_id` in `accessories`
- `location_id` in `asset_logs`
- `location_id` in `consumables`

**Comment by leobg** [comment]
Sorry for the long silence! Found the sucker finally! It was in asset_logs. And I think I figured why it happened. The user that asset is checked out used to be in that location I am trying to delete. The user since moved to another location along with the asset and everything. I have changed the user to the new location, as well as the asset. But the log entry about the checkout remained unchanged. I guess problem could have gone away if I checked in the assets, then check back out.
Not sure if it is a good idea to introduce a trigger when user gets location changed to automatically check-in/out all associated assets to address issues like that.
